quickjspp.hpp

```diff
@@ -1626,6 +1632,25 @@ class Context
     /** Callback triggered when a Promise rejection won't ever be handled */
     std::function<void(Value)> onUnhandledPromiseRejection;
 
+    /** Data type returned by the moduleLoader function */
+    struct ModuleData {
+        std::optional<std::string> source, url;
+        ModuleData()
+          : source(std::nullopt), url(std::nullopt)
+        {}
+        ModuleData(std::string source)
+          : source(std::move(source)), url(std::nullopt)
+        {}
+        ModuleData(std::string url, std::string source)
+          : source(std::move(source)), url(std::move(url))
+        {}
+    };
+
+    /** Function called to obtain the source of a nodule */
+    std::function<ModuleData(std::string_view)> moduleLoader = [](std::string_view filename){
+        return readFile(filename.data());
```

Comment:
std::string_view can be not null-terminated, e.g. after substr()

Reply:
yes, but the call to string_view of `moduleLoader` comes from a C style string, which is always null-terminated.

Reply:
You are right, thanks
```suggestion
        assert(filename.data()[filename.size()] == '\0'); // filename comes from a C style string
        return readFile(filename.data());
```

Reply:
In this case, the `std::string_view` will be constructed by `module_loader` from a C-style string, so it will always be null-terminated.

Reply:
With the change to use `std::filesystem` this assumption is no longer required.